src/XMLexport.cpp

```diff
@@ -533,6 +533,27 @@ void XMLexport::writeHost(Host* pHost, pugi::xml_node mudletPackage)
         host.append_child("mRoomSize").text().set(QString::number(pHost->mRoomSize, 'f', 1).toUtf8().constData());
     }
 
+    {
+        auto stopwatches = host.append_child("stopwatches");
+        QListIterator<int> itStopWatchId(pHost->getStopWatchIds());
+        while (itStopWatchId.hasNext()) {
+            auto stopWatchId = itStopWatchId.next();
+            auto pStopWatch = pHost->getStopWatch(stopWatchId);
+            if (pStopWatch->persistent()) {
+                auto stopwatch = stopwatches.append_child("stopwatch");
+                // Three QStrings used here are purely numeric so can be expressed in Latin1 encoding:
+                stopwatch.append_attribute("id") = QString::number(stopWatchId).toLatin1().constData();
+                if (pStopWatch->running()) {
+                    stopwatch.append_attribute("running") = "yes";
+                    stopwatch.append_attribute("effectiveStartDateTimeEpochMSecs") = QString::number(QDateTime::currentMSecsSinceEpoch() - pStopWatch->getElapsedMilliSeconds()).toLatin1().constData();
```

Comment:
why not simply use https://doc.qt.io/qt-5/qdatetime.html#toMSecsSinceEpoch?

Reply:
I have to think about that - to see if that is what I am actually calculating at that point, it *might* be. 

Ah, I've encapsulated the `stopWatch` class member that contains the value that would be needed (it is private) and - thought it is accessible through a debugging getter - it seems a better design choice to use the public API that I constructed. Not a terribly strong reason admittedly, but the overall design works for me - and since it has been a little while since I coded this it might take time for me to get back into it to see if that is a reasonable alternative.

Reply:
But in https://github.com/Mudlet/Mudlet/pull/2516/files/84d150ecdd1eb3bed3cc9599276ac6a1622dfa02#diff-ea6759c1012dd183514ddaded83fba1eR1783 you do access the `mEffectiveStartDateTime` directly.